### ISSUE TITLE
Worker Launch Template Invalid Resource Type

### ISSUE THREAD

**Issue by toneill818** [terraform-aws-modules/terraform-aws-eks#1406]
## Description

`resource_type = "network-interface" ` is not supported

## Versions
 - Terraform: 0.14.4
 - Provider(s):

 - Module: 17.0.1

## Reproduction
Create a cluster with `worker_groups_launch_template`

## Additional context
According to the documentation "network-interface" is not a supported resource_type. 
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#tag-specifications

Seems to have been added in PR #1362

**Comment by marcuz** [comment]
Same here:

```hcl
╷
│ Error: expected tag_specifications.2.resource_type to be one of [instance volume spot-instances-request elastic-gpu], got network-interface
│ 
│   with [REDACTED].module.[REDACTED].aws_launch_template.workers_launch_template[0],
│   on .terraform/modules/[REDACTED]/workers_launch_template.tf line 567, in resource "aws_launch_template" "workers_launch_template":
│  567:     resource_type = "network-interface"
│ 
╵
```

**Comment by barryib** [comment]
Fixed in [v17.0.2](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/tag/v17.0.2)

**Comment by marcuz** [comment]
Thank you @barryib! :raised_hands:
